src/delegate.rs

```diff
@@ -270,6 +276,18 @@ impl<T: LanguageServer> LanguageServerCore for Delegate<T> {
             Box::new(self.server.document_highlight(p))
         })
     }
+
+    fn signature_help(&self, params: Params) -> BoxFuture<Option<SignatureHelp>> {
+        self.delegate_request::<SignatureHelpRequest, _>(params, |p| {
+            Box::new(self.server.signature_help(p))
+        })
+    }
+
+    fn goto_implementation(&self, params: Params) -> BoxFuture<Option<GotoImplementationResponse>> {
```

Comment:
This method should be moved to immediately below `Delegate::goto_type_definition()`.